RELEASING.md

```diff
@@ -0,0 +1,50 @@
+# Releasing
+
+Currently the only external artefact for a release is the SQL runner image in quay.io.
+
+The automation builds and pushes images to quay on push to:
+1. main (image will be tagged as main eg quay.io/streamshub/flink-sql-runner:main )
+2. branches named release-** (image will be tagged as branch name eg release-0.0 will be tagged at quay.io/streamshub/flink-sql-runner:release-0.0)
```

Comment:
in strimzi we use `releas-0.0.x`, guess for example `0.0.2` will be build from `release-0.0`. I don't say we should name it in same way, it is just and example of another possibility.